Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -1468,6 +1468,8 @@ void LocalDOMWindow::setName(const AtomString& string)
         return;
 
     frame->tree().setSpecifiedName(string);
+    if (RefPtr page = frame->page())
+        page->chrome().frameNameChanged(frame->frameID(), string.string());
```

Comment:
FrameLoaderClient would be a slightly better place for this because it's scoped to the frame rather than the page.